cli/info.rs

```diff
@@ -1,404 +1,232 @@
-    let files = FileInfoDepFlatGraph::new(&module_graph);
-
-    let info = Self {
-      module: module_specifier.to_string(),
-      local: local_filename,
-      file_type,
-      compiled: compiled_filename,
-      map: map_filename,
-      dep_count,
-      deps,
-      total_size,
-      files,
-    };
+/// Serialize a module specifier to a simple string representation.
+fn serialize_specifier<S>(ms: &ModuleSpecifier, s: S) -> Result<S::Ok, S::Error>
+where
+  S: Serializer,
+{
+  s.serialize_str(&ms.to_string())
+}
 
-    Ok(info)
+/// Serialize a vector of module specifiers.
+fn serialize_specifier_vec<S>(
+  v: &[ModuleSpecifier],
+  s: S,
+) -> Result<S::Ok, S::Error>
+where
+  S: Serializer,
+{
+  let mut seq = s.serialize_seq(Some(v.len()))?;
+  for element in v {
+    seq.serialize_element(&element.to_string())?;
   }
+  seq.end()
 }
 
-/// Counts the number of dependencies in the graph.
-///
-/// We are counting only the dependencies that are not http redirects to other files.
-fn get_unique_dep_count(graph: &ModuleGraph) -> usize {
-  graph.iter().fold(
-    0,
-    |acc, e| {
-      if e.1.redirect.is_none() {
-        acc + 1
-      } else {
-        acc
-      }
-    },
-  )
+/// The core structure representing information about a specific "root" file in
+/// a module graph.  This is used to represent information as part of the `info`
+/// subcommand.
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ModuleGraphInfo {
+  pub compiled: Option<PathBuf>,
+  pub dep_count: usize,
+  #[serde(serialize_with = "serialize_media_type")]
+  pub file_type: MediaType,
+  pub files: ModuleInfoMap,
+  #[serde(skip_serializing)]
+  pub info: ModuleInfo,
+  pub local: PathBuf,
+  pub map: Option<PathBuf>,
+  #[serde(serialize_with = "serialize_specifier")]
+  pub module: ModuleSpecifier,
+  pub total_size: usize,
 }
 
-impl std::fmt::Display for ModuleDepInfo {
-  fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-    f.write_fmt(format_args!("{} {}\n", colors::bold("local:"), self.local))?;
-    f.write_fmt(format_args!(
-      "{} {}\n",
-      colors::bold("type:"),
-      self.file_type
-    ))?;
+impl fmt::Display for ModuleGraphInfo {
+  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+    writeln!(
+      f,
+      "{} {}",
+      colors::bold("local:"),
+      self.local.to_string_lossy()
+    )?;
+    writeln!(f, "{} {}", colors::bold("type:"), self.file_type)?;
     if let Some(ref compiled) = self.compiled {
-      f.write_fmt(format_args!(
-        "{} {}\n",
+      writeln!(
+        f,
+        "{} {}",
         colors::bold("compiled:"),
-        compiled
-      ))?;
+        compiled.to_string_lossy()
+      )?;
     }
     if let Some(ref map) = self.map {
-      f.write_fmt(format_args!("{} {}\n", colors::bold("map:"), map))?;
+      writeln!(f, "{} {}", colors::bold("map:"), map.to_string_lossy())?;
     }
-
-    f.write_fmt(format_args!(
-      "{} {} unique {}\n",
```

Comment:
It's odd to me that this module graph refactor is effecting the display. Are these changes unrelated edits?

Reply:
I could choose to not refactor the display to more idiomatic Rust, but that seems non-sensical.  I really don't want to step around unmaintainable code when there is an opportunity to refactor it.

Reply:
Make its it easier to understand during complex refactors if only the necessary bits are touched - but no big deal